datasource/src/main/scala/quasar/physical/mongo/Mongo.scala

```diff
@@ -81,10 +84,17 @@ class Mongo[F[_]: MonadResourceErr : ConcurrentEffect] private[Mongo](client: Mo
     } yield res)
   }
 
-
-
   def databases: Stream[F, Database] =
-    observableAsStream(client.listDatabaseNames, DefaultQueueSize).map(Database(_))
+    observableAsStream(client.listDatabaseNames, DefaultQueueSize)
+      .map(Database(_))
+      .handleErrorWith { _ match {
```

Comment:
FYI, the `_ match {` is unnecessary. Match bodies are actually partial functions, which are themselves function values.